assembl/views/api2/content_source.py

```diff
@@ -55,6 +61,9 @@ def fetch_posts(request):
             raise HTTPUnauthorized("Only discussion administrator\
                                    can "+'and'.join(requested))
 
+    if force_restart:
```

Comment:
Is the `fetch_post` method responsible for calling to reset all errors in the `ContentSource`? It's a bit spaghetti, non? Could be further down the call chain, check for the force_restart and have the `ContentSource` take care of its own `reset_errors()`. A thought, I'm not picky.